### PR TITLE
Fix `ImmutableTypeCloner` not setting the correct origin for virtual symbols

### LINKED ISSUE #35629: Packages loaded from BIR show anon types in module's symbols
**Description:**
$subject causes #35334. Seems like some of the virtual symbols that are load from BIR are not set with the `origin=VIRTUAL` causing those symbols to appear in `module.allSymbols()`.

Note that this behavior is observed in distributions, but not when a package is pulled from central.

**Steps to reproduce:**
Type `http:<cursor>` in any context and have a look at the suggestions. Some anon types and intersection types are present in the list.

**Affected Versions:**
2201.0.0, 2201.0.1, 2201.0.2

### PR DESCRIPTION
**Pull request by rdulmina**
## Purpose
$title

Fixes #35629

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/util/ImmutableTypeCloner.java [hunk 1]
```diff
@@ -567,7 +567,7 @@ private static BIntersectionType defineImmutableRecordType(Location pos, BRecord
         BRecordTypeSymbol recordSymbol =
                 Symbols.createRecordSymbol(origRecordType.tsymbol.flags | Flags.READONLY,
                         getImmutableTypeName(names,  getSymbolFQN(origRecordType.tsymbol)),
-                        pkgID, null, env.scope.owner, pos, SOURCE);
+                        pkgID, null, env.scope.owner, pos, origRecordType.tsymbol.origin);
```
Comment (KavinduZoysa) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/util/ImmutableTypeCloner.java:570]: In this method `origRecordType.tsymbol` is used in several places, shall we move that into a separate variable?

**Review by KRVPerera — APPROVED** [review]

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/util/ImmutableTypeCloner.java (hunk 1 above)
Comment (rdulmina) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/util/ImmutableTypeCloner.java]: Updated

**Review by KRVPerera — APPROVED** [review]

**Review by KavinduZoysa — DISMISSED** [review]

**Comment by KRVPerera** [comment]
Shall we change the branch to master?

**Comment by KRVPerera** [comment]
Shall we change the branch to master?

**Review by KavinduZoysa — APPROVED** [review]

**Review by KRVPerera — APPROVED** [review]

**Pull request merged**